spec/orbital/remote/shared_examples_for_visa_mit_cit_framework.rb

```diff
@@ -0,0 +1,96 @@
+    payment_response.status.should ==:PROCESSED
+    payment_response.amount.should == @amount
+    payment_response.transaction_type.should == :AUTHORIZE
+    find_value_from_properties(payment_response.properties, 'processorResponse').should == '100'
+  end
+
+  it 'should not able to do successful authorization with non-matching mit ref id' do
+    properties = merge_extra_properties(@properties, [build_property('mit_cit_type', 'CGEN'),
+                                                      build_property('credential_on_file', true)])
+    @plugin.authorize_payment(@pm.kb_account_id, @kb_payment.id, @kb_payment.transactions[0].id, @pm.kb_payment_method_id, @amount, @currency, properties, @call_context)
+
+    properties = merge_extra_properties(@properties, [build_property('mit_cit_type', 'MRSB'),
+                                                      build_property('credential_on_file', true),
+                                                      build_property('mit_ref_trx_id', 'test_id')])
+    payment_response = @plugin.authorize_payment(@pm.kb_account_id, @kb_payment.id, @kb_payment.transactions[0].id, @pm.kb_payment_method_id, @amount, @currency, properties, @call_context)
+    payment_response.status.should == :ERROR
+    payment_response.gateway_error.should ==' MIT: Original transaction id is mandatory for merchant initiated transactions.'
+  end
+
+  it 'should be able to do successful purchase with mit ref id' do
+    properties = merge_extra_properties(@properties, [build_property('mit_cit_type', 'CGEN'),
+                                                      build_property('credential_on_file', true)])
+    payment_response = @plugin.purchase_payment(@pm.kb_account_id, @kb_payment.id, @kb_payment.transactions[0].id, @pm.kb_payment_method_id, @amount, @currency, properties, @call_context)
+    mit_trx_id = find_value_from_properties(payment_response.properties, 'mit_received_transaction_id')
+    mit_trx_id.should_not be_nil
+
+    properties = merge_extra_properties(@properties, [build_property('mit_cit_type', 'MRSB'),
+                                                      build_property('credential_on_file', true),
+                                                      build_property('mit_ref_trx_id', @kb_payment.transactions[0].id)])
+    payment_response = @plugin.purchase_payment(@pm.kb_account_id, @kb_payment.id, @kb_payment.transactions[0].id, @pm.kb_payment_method_id, @amount, @currency, properties, @call_context)
+    payment_response.status.should eq(:PROCESSED), payment_response.gateway_error
+    payment_response.amount.should == @amount
+    payment_response.transaction_type.should == :PURCHASE
+    find_value_from_properties(payment_response.properties, 'processorResponse').should == '100'
+  end
+
+  def merge_extra_properties(properties, extra_properties)
```

Comment:
This helper class could help maybe: https://github.com/killbill/killbill-plugin-framework-ruby/blob/e789d92f2253d3a28e596321fad9a6720592584e/lib/killbill/helpers/properties_helper.rb.

Reply:
Switched to the method in helper.